Helper/DocumentImporter.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+
+namespace EMS\CoreBundle\Helper;
+
+
+use Doctrine\ORM\EntityManager;
+use EMS\CoreBundle\Elasticsearch\Bulker;
+use EMS\CoreBundle\Entity\ContentType;
+use EMS\CoreBundle\Entity\Environment;
+use EMS\CoreBundle\Entity\Revision;
+use EMS\CoreBundle\Exception\CantBeFinalizedException;
+use EMS\CoreBundle\Form\Form\RevisionType;
+use EMS\CoreBundle\Repository\ContentTypeRepository;
+use EMS\CoreBundle\Repository\RevisionRepository;
+use EMS\CoreBundle\Service\DataService;
+use Symfony\Component\Form\FormFactoryInterface;
+
+class DocumentImporter
+{
+    /** @var ContentType */
+    private $contentType;
+    /** @var string */
+    private $contentTypeName;
+    /** @var DataService */
+    private $dataService;
+    /** @var string */
+    private $lockUser;
+    /** @var bool */
+    private $rawImport;
+    /** @var EntityManager */
+    private $entityManager;
+    /** @var RevisionRepository */
+    private $revisionRepository;
+    /** @var FormFactoryInterface */
+    private $formFactory;
+    /** @var bool */
+    private $indexInDefaultEnv;
+    /** @var bool */
+    private $signData;
+    /** @var Bulker */
+    private $bulker;
+    /** @var ContentTypeRepository */
+    private $contentTypeRepository;
+    /** @var Environment */
+    private $environment;
+    /** @var int */
+    private $bulkSize;
+    /** @var string  */
+    private $instanceId;
+    /** @var bool */
+    private $finalize;
+    /** @var bool */
+    private $force;
+
+    public function __construct(DataService $dataService, EntityManager $entityManager, FormFactoryInterface $formFactory, Bulker $bulker, string $instanceId, string $contentTypeName, string $lockUser, bool $rawImport, bool $signData, bool $indexInDefaultEnv, int $bulkSize, bool $finalize, bool $force)
+    {
+        $this->contentTypeName = $contentTypeName;
+        $this->indexInDefaultEnv = $indexInDefaultEnv;
+        $this->instanceId = $instanceId;
+        $this->signData = $signData;
+        $this->lockUser = $lockUser;
+        $this->rawImport = $rawImport;
+        $this->bulkSize = $bulkSize;
+        $this->finalize = $finalize;
+        $this->dataService = $dataService;
+        $this->bulker = $bulker;
+        $this->entityManager = $entityManager;
+        $this->formFactory = $formFactory;
+        $this->force = $force;
```

Comment:
* move services to the ImportService class.
* move options to a contextual object for the ImportService class

Reply:
1e8555d195f4caf512272f5f4efed1125cc64771